modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/preupdatepassword/PreUpdatePasswordActionFailureTestCase.java

```diff
@@ -0,0 +1,344 @@
+        this.actionResponse = actionResponse;
+        this.expectedPasswordUpdateResponse = expectedPasswordUpdateResponse;
+    }
+
+    @DataProvider(name = "testExecutionContextProvider")
+    public static Object[][] getTestExecutionContext() throws IOException, URISyntaxException {
+
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_USER, new ActionResponse(200,
+                        FileUtils.readFileInClassPathAsString("actions/response/failure-response.json")),
+                        new ExpectedPasswordUpdateResponse("400", "Some failure reason. " +
+                                "Some description")},
+                {TestUserMode.TENANT_USER, new ActionResponse(200,
+                        FileUtils.readFileInClassPathAsString("actions/response/failure-response.json")),
+                        new ExpectedPasswordUpdateResponse("400", "Some failure reason. " +
+                                "Some description")},
+                {TestUserMode.SUPER_TENANT_USER, new ActionResponse(500,
+                        FileUtils.readFileInClassPathAsString("actions/response/error-response.json")),
+                        new ExpectedPasswordUpdateResponse("500", "Error while updating " +
+                                "attributes of user")},
+                {TestUserMode.TENANT_USER, new ActionResponse(500,
+                        FileUtils.readFileInClassPathAsString("actions/response/error-response.json")),
+                        new ExpectedPasswordUpdateResponse("500", "Error while updating " +
+                                "attributes of user")},
+        };
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        Utils.getMailServer().purgeEmailFromAllMailboxes();
+        super.init(userMode);
+
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+
+        application = addApplicationWithGrantType(CLIENT_CREDENTIALS_GRANT_TYPE);
+        applicationId = application.getId();
+        OpenIDConnectConfiguration oidcConfig = getOIDCInboundDetailsOfApplication(applicationId);
+        clientId = oidcConfig.getClientId();
+        clientSecret = oidcConfig.getClientSecret();
+
+        UserObject userInfo = new UserObject();
+        userInfo.setUserName(TEST_USER1_USERNAME);
+        userInfo.setPassword(TEST_USER_PASSWORD);
+        userInfo.setName(new Name().givenName(TEST_USER_GIVEN_NAME));
+        userInfo.getName().setFamilyName(TEST_USER_LASTNAME);
+        userInfo.addEmail(new Email().value(TEST_USER_EMAIL));
+        userId = scim2RestClient.createUser(userInfo);
+
+        updatePasswordRecoveryFeatureStatus(true);
+        updateAdminPasswordResetRecoveryEmailFeatureStatus(true);
+        updateAdminInitiatedPasswordResetEmailFeatureStatus(true);
+
+        actionId = createPreUpdatePasswordAction(ACTION_NAME, ACTION_DESCRIPTION);
+
+        actionsMockServer = new ActionsMockServer();
```

Comment:
Let's move this to a method in (e.g., `startWireMockServer()`) ActionBaseTestCase and call that here.